addons/hr_timesheet/wizard/hr_employee_delete_wizard.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+
+
+class HrEmployeDeleteWizard(models.TransientModel):
+    _name = 'hr.employee.delete.wizard'
+    _description = 'Employee Delete Wizard'
+
+    employee_count = fields.Integer(string='Employee Count', compute='_compute_employee_data')
+    employee_ids = fields.Many2many('hr.employee', string='Employees')
+    has_active_employee = fields.Boolean(string='Has Active Employee', compute='_compute_employee_data')
+    has_timesheet = fields.Boolean(string='Has Timesheet', compute='_compute_employee_data')
+
+    def _compute_employee_data(self):
+        self = self.with_context(active_test=False)
+        timesheet_read_group = self.env['account.analytic.line']._read_group([
+            ('employee_id', 'in', self.employee_ids.ids)],
+            ['employee_id'], ['employee_id']
+        )
+        timesheet_employee_map = set(employee_info['employee_id'][0] for employee_info in timesheet_read_group)
+        for wizard in self:
+            wizard.employee_count = len(wizard.employee_ids)
+            wizard.has_active_employee = wizard.with_context(active_test=True).employee_ids
+            wizard.has_timesheet = timesheet_employee_map & set(wizard.employee_ids.ids)
+
+    def action_archive(self):
+        if self.employee_count == 1:
+            return {
+                'name': _('Employee Termination'),
+                'type': 'ir.actions.act_window',
+                'res_model': 'hr.departure.wizard',
+                'views': [[False, 'form']],
+                'view_mode': 'form',
+                'target': 'new',
+                'context': {
+                    'active_id': self.employee_ids.id,
+                    'toggle_active': True,
+                }
+            }
+        return self.employee_ids.toggle_active()
```

Comment:
Small detail (I know I can be a pain in the butt) :
```suggestion
        if self.employee_count != 1:
            return self.employee_ids.toggle_active()
        return {
            'name': _('Employee Termination'),
            'type': 'ir.actions.act_window',
            'res_model': 'hr.departure.wizard',
            'views': [[False, 'form']],
            'view_mode': 'form',
            'target': 'new',
            'context': {
                'active_id': self.employee_ids.id,
                'toggle_active': True,
            }
        }
```
Gaining an indentation. Up to you!